packages/next/src/client/components/router-reducer/reducers/server-action-reducer.ts

```diff
@@ -98,24 +102,24 @@ async function fetchServerAction(
       }
     )
 
-    // if it was a redirection, then result is just a regular RSC payload
     if (location) {
-      const [, payload] = response
+      // if it was a redirection, then result is just a regular RSC payload
+      const [, actionFlightData] = (response as any) ?? []
```

Comment:
do we need the any here?

Reply:
Because the types are inconsistent, when you call `redirect()` the server sends a `FlightData` but when you don't it sends `[UserFnReturnValue, FlightData]`, so these don't line up correctly, but changing the server to return the same result is non-trivial because of the way redirect() is implemented.